### PR TITLE
[Backport][ipa-4-6] Test for removing a subgroup

### PR DESCRIPTION
**Pull request by ssidhaye**
Problem description:
Removing an IPA sub-group should NOT remove the members
from indirect parent that also belong to other subgroups

The test:
A user and three groups are created a,b,c
'c' should be a child of 'b' so that you have a->b->c

user is direct member of 'a' and as a result member of 'b'
and 'c'. Now when one adds a direct membership to 'b' nothing will
change.

If one removes the direct membership to 'b' again,
nothing should change as well

Pagure Link: https://pagure.io/SSSD/sssd/issue/3636

Manual backport of #4544

Signed-off-by: Sumedh Sidhaye <ssidhaye@redhat.com>

### CODE REVIEW

File: ipatests/test_integration/test_sssd.py [hunk 1]
```diff
@@ -348,3 +350,179 @@ def get_cache_update_time(obj_kind, obj_name):
         finally:
             sssd_conf_backup.restore()
             tasks.clear_sssd_cache(self.master)
+
+
+@pytest.skipif("SSSD fix is not available on Fedora 27")
```
Comment (flo-renaud) [inline ipatests/test_integration/test_sssd.py:355]: The annotation should be `@pytest.mark.skip(reason=...")` or `@pytest.mark.skipif(condition, reason="...")`

**Comment by flo-renaud** [comment]
Backport LGTM.

@wladich do you prefer to have the @pytest.mark.skip annotation in a separate commit?

**Comment by wladich** [comment]
@flo-renaud 
Yes, separate commit is highly preferred as it will help to avoid manual editing of the commit during backporting.

**Comment by ssidhaye** [comment]
> @flo-renaud
> Yes, separate commit is highly preferred as it will help to avoid manual editing of the commit during backporting.

ok @wladich I'll move it to separate commit.

File: ipatests/test_integration/test_sssd.py (hunk 1 above)
Comment (ssidhaye) [reply ipatests/test_integration/test_sssd.py]: fixed it

**Comment by wladich** [comment]
@ssidhaye 
LGTM

**Comment by rcritten** [comment]
LGTM, please remove the temporary commit.

**Comment by rcritten** [comment]
Sorry, just noticed the commit message still doesn't match. It should look like:

Test for removing a subgroup

Problem description: 
Removing ...

I"m not trying to be pedantic but it is sometimes important when looking for changes between branches for the commit messages to match up.

No need to re-add a temporary commit as this just affects the message.

**Comment by flo-renaud** [comment]
ipa-4-6:

* 5f5174b0c1d058d0d7926bcad7689e2166a5776f Test for removing a subgroup
* 6a1dd26dc6a62e37b501068634f1a6804a495ab5 Added pytest.mark.skip as SSSD fix for https://pagure.io/SSSD/sssd/issue/3636 is not available in Fedora 27

**Pull request closed without merging**